crates/sui-types/src/authenticator_state.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use fastcrypto_zkp::bn254::zk_login::{JwkId, JWK};
+use move_core_types::{account_address::AccountAddress, ident_str, identifier::IdentStr};
+use serde::{Deserialize, Serialize};
+
+use crate::dynamic_field::get_dynamic_field_from_store;
+use crate::error::{SuiError, SuiResult};
+use crate::storage::ObjectStore;
+use crate::{id::UID, SUI_AUTHENTICATOR_STATE_OBJECT_ID, SUI_FRAMEWORK_ADDRESS};
+
+pub const AUTHENTICATOR_STATE_MODULE_NAME: &IdentStr = ident_str!("authenticator_state");
+pub const AUTHENTICATOR_STATE_STRUCT_NAME: &IdentStr = ident_str!("AuthenticatorState");
+pub const RESOLVED_SUI_AUTHENTICATOR_STATE: (&AccountAddress, &IdentStr, &IdentStr) = (
+    &SUI_FRAMEWORK_ADDRESS,
+    AUTHENTICATOR_STATE_MODULE_NAME,
+    AUTHENTICATOR_STATE_STRUCT_NAME,
+);
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AuthenticatorState {
+    pub id: UID,
+    pub version: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AuthenticatorStateInner {
+    pub version: u64,
+
+    /// List of currently active JWKs.
+    pub active_jwks: Vec<ActiveJwk>,
```

Comment:
Do we have a limit on this size? Any chance now/in the future we could come close to this limit?

Reply:
There is currently no limit on how many we store on chain. The fetch side is limited to 100 JWKs per provider per hour.

Suppose that one oauth provider goes rogue and tries to inundate us with bogus keys. That would be 100 new keys every hour. Google keys are about 500 bytes each. Eventually we will only keep keys from the current and prior epoch, so we would have about 2.5MB of keys in storage.

Of course, they could also send us extremely large keys. I'm adding code now to limit the size of ID + key to 4096 bytes.